From: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
Date: Sun, 17 Apr 2016 17:53:46 +0200
Subject: [PATCH] target Ubuntu: Blacklist rtc-efi if not supported.

Older ThunderX and possibly other UEFI-enabled targets do not support
rtc-efi properly, so they end up filling dmesg with useless
complaints about not being able to read system time.

Target OS (Ubuntu) deployment already has a snippet that blacklists
i2c_piix4 in certain scenarios, so provide a similar test for rtc_efi.

update-initramfs is already invoked from i2c_piix4 snippet, so ommit
explicitly calling it in rtc_efi fragment.
---
 deployment/puppet/cobbler/manifests/snippets.pp                          | 1 +
 deployment/puppet/cobbler/templates/preseed/ubuntu-1404.preseed.erb      | 1 +
 .../puppet/cobbler/templates/snippets/ubuntu_blacklist_rtc_efi.erb       | 1 +
 3 files changed, 3 insertions(+)
 create mode 100644 deployment/puppet/cobbler/templates/snippets/ubuntu_blacklist_rtc_efi.erb

diff --git a/deployment/puppet/cobbler/manifests/snippets.pp b/deployment/puppet/cobbler/manifests/snippets.pp
index 46c76d0..5ab9a17 100644
--- a/deployment/puppet/cobbler/manifests/snippets.pp
+++ b/deployment/puppet/cobbler/manifests/snippets.pp
@@ -58,6 +58,7 @@ class cobbler::snippets {
   cobbler_snippet {"centos_static_net":}
   cobbler_snippet {"ofed_install_with_sriov":}
   cobbler_snippet {"ubuntu_authorized_keys":}
+  cobbler_snippet {"ubuntu_blacklist_rtc_efi":}
   cobbler_snippet {"ubuntu_blacklist_i2c_piix4":}
   cobbler_snippet {"ubuntu_disable_pxe":}
   cobbler_snippet {"ubuntu_puppet_config":}
diff --git a/deployment/puppet/cobbler/templates/preseed/ubuntu-1404.preseed.erb b/deployment/puppet/cobbler/templates/preseed/ubuntu-1404.preseed.erb
index 65c6c69..8e17a0e 100644
--- a/deployment/puppet/cobbler/templates/preseed/ubuntu-1404.preseed.erb
+++ b/deployment/puppet/cobbler/templates/preseed/ubuntu-1404.preseed.erb
@@ -177,6 +177,7 @@ echo target > /target/etc/nailgun_systemtype && \
 $SNIPPET('ubuntu_authorized_keys')
 sed -i -e "/^\s*GSSAPICleanupCredentials yes/d" -e "/^\s*GSSAPIAuthentication yes/d" -e "s/.*PasswordAuthentication\ .*/PasswordAuthentication\ no/g" -e "/UseDNS/d" /target/etc/ssh/sshd_config && \
 echo "UseDNS no" >> /target/etc/ssh/sshd_config && \
+$SNIPPET('ubuntu_blacklist_rtc_efi')
 $SNIPPET('ubuntu_blacklist_i2c_piix4')
 $SNIPPET('ubuntu_static_net')
 $SNIPPET('ofed_install_with_sriov')
diff --git a/deployment/puppet/cobbler/templates/snippets/ubuntu_blacklist_rtc_efi.erb b/deployment/puppet/cobbler/templates/snippets/ubuntu_blacklist_rtc_efi.erb
new file mode 100644
index 0000000..902f142
--- /dev/null
+++ b/deployment/puppet/cobbler/templates/snippets/ubuntu_blacklist_rtc_efi.erb
@@ -0,0 +1 @@
+( /bin/cat /sys/class/rtc/rtc0/time > /dev/null 2>&1 || echo "blacklist rtc_efi" >> /target/etc/modprobe.d/blacklist-rtc_efi.conf ) && \
